rm/rm-node-updater/src/main/java/org/ow2/proactive/resourcemanager/updater/RMNodeUpdater.java

```diff
@@ -46,113 +74,496 @@
  * When an agent detects it and restart the node it just proxies the request to RMNodeStarer.
  *
  */
-public class RMNodeUpdater {
+public class RMNodeUpdater extends RMNodeStarter {
 
+    /**
+     * Url used to download node.jar
+     */
     private static final String NODE_URL_PROPERTY = "node.jar.url";
 
-    private static final String NODE_JAR_PROPERTY = "node.jar.saveas";
+    static final String OPTION_NODE_URL = "nju";
+
+    public static final String OPTION_NODE_JAR_URL_NAME = "nodeJarUrl";
+
+    public static final String OPTION_NODE_JAR_SAVE_AS_NAME = "nodeJarSaveAs";
+
+    protected String nodeJarUrl;
+
+    /**
+     * Local path where the node jar should be stored
+     */
+    private static final String NODE_JAR_SAVEAS_PROPERTY = "node.jar.saveas";
+
+    static final String OPTION_NODE_SAVEAS = "njs";
+
+    protected String nodeJarSaveAs;
+
+    /**
+     * If true, then the updater never terminates, it will
+     */
+    private static final String NODE_UPDATER_AUTOMATIC_RELAUNCH = "node.updater.automatic.relaunch";
+
+    static final String OPTION_NODE_AUTOMATIC = "nja";
+
+    protected boolean automaticRelaunch = false;
+
+    /**
+     * optional One-Jar property, path used to expand librairies
+     */
+    private static final String ONEJAR_EXPAND_DIR_PROPERTY = "one-jar.expand.dir";
+
+    /**
+     * Java Property prefix used to enter extra java command line options
+     * For example -Xmn256m can be configured by using -DXtraOption1=Xmn256m
+     */
+    public static final String XTRA_OPTION = "XtraOption";
+
+    public static final String LAST_DOT_AND_AFTER = "\\.[^\\.]*$";
+
+    private static final Logger logger = initLogger();
 
-    private static final String TMP_DIR_PROPERTY = "java.io.tmpdir";
+    public static final int NUMBER_OF_ATTEMPTS = 10;
 
-    public static final String ONE_JAR_JAR_PATH = "one-jar.jar.path";
+    public RMNodeUpdater() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)